src/geometry.F90

```diff
@@ -688,427 +629,26 @@ subroutine distance_to_boundary(p, dist, surface_crossed, lattice_translation, &
       ! =======================================================================
       ! FIND MINIMUM DISTANCE TO SURFACE IN THIS CELL
 
-      SURFACE_LOOP: do i = 1, cl % n_surfaces
-
-        ! copy local coordinates of particle
-        x = p % coord(j) % xyz(1)
-        y = p % coord(j) % xyz(2)
-        z = p % coord(j) % xyz(3)
-
-        ! check for coincident surface -- note that we can't skip the
-        ! calculation in general because a particle could be on one side of a
-        ! cylinder and still have a positive distance to the other
-
-        index_surf = cl % surfaces(i)
-        if (index_surf == p % surface) then
-          on_surface = .true.
-        else
-          on_surface = .false.
-        end if
+      SURFACE_LOOP: do i = 1, size(cl%region)
+        ! check for operators
+        index_surf = cl%region(i)
+        coincident = (index_surf == p % surface)
 
         ! check for operators
```

Comment:
Replace "check for operators" with something a little more descriptive like "Ignore this token if it corresponds to an operator rather than a region."